winpr/libwinpr/thread/thread.c

```diff
@@ -558,7 +558,7 @@ BOOL ThreadCloseHandle(HANDLE handle)
 
 		if ((thread->started) && (WaitForSingleObject(thread, 0) != WAIT_OBJECT_0))
 		{
-			WLog_ERR(TAG, "Thread running, setting to detached state!");
+			WLog_DBG(TAG, "Thread running, setting to detached state!");
 			thread->detached = TRUE;
 			pthread_detach(thread->thread);
```

Comment:
Not related to this PR, but I'm wondering if there's a possibility for a race condition here if the thread terminates just after the `WaitForSingleObject` call (what's the behaviour of `pthread_detach` on a terminated thread ?)

Reply:
According to [this](https://stackoverflow.com/questions/51882300/calling-pthread-detach-for-an-already-exited-thread?answertab=votes#tab-top) I don't think that there is a problem with calling pthread_detach, but I do think that the other branch that calls `cleanup_handle` won't be executed (?)

Reply:
@hardening @kubistika no, the `cleanup_handle` is also called for threads exiting prematurely (see `thread_launcher`)